tests/Microsoft.Identity.Test.Unit/CoreTests/RegionDiscoveryProviderTests.cs

```diff
@@ -137,6 +137,20 @@ public async Task ResponseFromUserProvidedRegionDifferentFromRegionDetectedAsync
             regionDiscoveryProvider.Clear();
         }
 
+        [TestMethod]
+        public async Task SuccessfulResponseFromRegionalizedAuthorityAsync()
+        {
+            var regionalizedAuthority = new Uri($"https://{TestConstants.Region}.login.microsoft.com/common/");
+
+            Environment.SetEnvironmentVariable(TestConstants.RegionName, TestConstants.Region);
+            
+            // In the instance discovery flow, TryGetMetadataAsync is always called with a known authority first, then with regionalized.
+            var t = await _regionDiscoveryProvider.TryGetMetadataAsync(new Uri("https://login.microsoftonline.com/common/"), _testRequestContext).ConfigureAwait(false);
+            InstanceDiscoveryMetadataEntry regionalMetadata = await _regionDiscoveryProvider.TryGetMetadataAsync(regionalizedAuthority, _testRequestContext).ConfigureAwait(false);
+
+            ValidateInstanceMetadata(regionalMetadata);
+        }
+
```

Comment:
what's the e2e scenario to reproduce this, from a customer perspective? this is a good test, but how does a customer get into this situation? would be interesting to have a test that covers that scenario, without forcing the 2 calls to the instance discovery endpoint separately.

also, clear the discoveryProviders at the end. remove unused variable `t` or assert something with it.

Reply:
The scenario is when the app is run the first time and regionalization is turned on and the instance cache is empty. In end-to-end flow, `TryGetMetadataAsync` ends up being called twice - once for known authority and once for regionalized. Although, this might make more sense to convert this into an integration test.

In MS Test, the constructor and instance variables are run before each test method. I verified that instance cache is clear before each test. I did add code to clear the cache in test cleanup anyway, though.